File: app/src/main/java/com/wa2c/android/medoly/plugin/action/twitter/EditActivity.java
package com.wa2c.android.medoly.plugin.action.twitter;

import android.app.ActionBar;
import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.text.Editable;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.Toast;

import com.wa2c.android.medoly.plugin.action.twitter.dialog.ConfirmDialogFragment;
import com.wa2c.android.medoly.plugin.action.twitter.dialog.InsertPropertyDialogFragment;
import com.wa2c.android.medoly.plugin.action.twitter.dialog.PropertyPriorityDialogFragment;


public class EditActivity extends Activity {

    /** 設定。 */
    private SharedPreferences sharedPreferences;
    /** 編集テキスト。 */
    private EditText contentEditText;
    /** アルバムアート挿入 */
    private CheckBox insertAlbumArtCheckBox;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        contentEditText = (EditText)findViewById(R.id.contentEditText);
        insertAlbumArtCheckBox = (CheckBox)findViewById(R.id.insertAlbumArtCheckBox);

        // アクションバー
        ActionBar actionBar = getActionBar();
        if (actionBar != null) {
            actionBar.setDisplayShowHomeEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowTitleEnabled(true);
        }

        // アルバムアート追加
        insertAlbumArtCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                sharedPreferences.edit().putBoolean(getString(R.string.prefkey_content_album_art), isChecked).apply();
            }
        });

        // 挿入
        findViewById(R.id.insertButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View v) {
                InsertPropertyDialogFragment dialogFragment = InsertPropertyDialogFragment.newInstance();
                dialogFragment.setOnItemSelectListener(new InsertPropertyDialogFragment.ItemSelectListener() {
                    @Override
                    public void onItemSelect(String insertString) {
                        int index1 = contentEditText.getSelectionStart();
                        int index2 = contentEditText.getSelectionEnd();
                        int start = Math.min(index1, index2);
                        int end = Math.min(index1, index2);
                        Editable editable = contentEditText.getText();
                        if (start > 0 && editable.charAt(start - 1) != ' ') {
                            insertString = " " + insertString; // スペース挿入
                        }
                        editable.replace(start, end, insertString);
                    }
                });
                dialogFragment.show(EditActivity.this);
            }
        });

        // 優先度
        findViewById(R.id.priorityButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View v) {
                PropertyPriorityDialogFragment dialogFragment = PropertyPriorityDialogFragment.newInstance();
//                dialogFragment.setOnItemSelectListener(new InsertPropertyDialogFragment.ItemSelectListener() {
//                    @Override
//                    public void onItemSelect(String insertString) {
//                        int index1 = contentEditText.getSelectionStart();
//                        int index2 = contentEditText.getSelectionEnd();
//                        int start = Math.min(index1, index2);
//                        int end = Math.min(index1, index2);
//                        Editable editable = contentEditText.getText();
//                        if (start > 0 && editable.charAt(Math.min(start, end) - 1) == ' ') {
//                            insertString = " " + insertString; // スペース挿入
//                        }
//                        editable.replace(start, end, insertString);
//                    }
//                });
                dialogFragment.show(EditActivity.this);
            }
        });


        // 初期化
        findViewById(R.id.initializeButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View v) {
                ConfirmDialogFragment dialogFragment = ConfirmDialogFragment.newInstance("確認", "確認");
                dialogFragment.setClickListener(new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (which == DialogInterface.BUTTON_POSITIVE) {
                            contentEditText.setText(getString(R.string.format_content_default));
                            insertAlbumArtCheckBox.setChecked(true);
                        }
                    }
                });
                dialogFragment.show(EditActivity.this);
            }
        });

        contentEditText.setText(sharedPreferences.getString(getString(R.string.prefkey_content_format), getString(R.string.format_content_default)));
        insertAlbumArtCheckBox.setChecked(sharedPreferences.getBoolean(getString(R.string.prefkey_content_album_art), true));
    }

    /**
     * onStopイベント処理。 c
     */
    @Override
    public void onStop() {
        super.onStop();

        // テキスト保存
        sharedPreferences.edit().putString(getString(R.string.prefkey_content_format), contentEditText.getText().toString()).apply();

     }

    /**
     * onOptionsItemSelected event.
     */
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/tweet/activity/EditActivity.kt
package com.wa2c.android.medoly.plugin.action.tweet.activity

import android.content.DialogInterface
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.wa2c.android.medoly.plugin.action.tweet.R
import com.wa2c.android.medoly.plugin.action.tweet.databinding.ActivityEditBinding
import com.wa2c.android.medoly.plugin.action.tweet.dialog.ConfirmDialogFragment
import com.wa2c.android.medoly.plugin.action.tweet.dialog.InsertPropertyDialogFragment
import com.wa2c.android.medoly.plugin.action.tweet.dialog.PropertyPriorityDialogFragment
import com.wa2c.android.prefs.Prefs


/**
 * Edit activity
 */
class EditActivity : AppCompatActivity() {

    private lateinit var  prefs: Prefs
    private lateinit var binding: ActivityEditBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        prefs = Prefs(this)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_edit)

        // Action bar
        supportActionBar?.let {
            it.setDisplayShowHomeEnabled(true)
            it.setDisplayHomeAsUpEnabled(true)
            it.setDisplayShowTitleEnabled(true)
        }

        // Insert album art button
        binding.insertAlbumArtCheckBox.setOnCheckedChangeListener { _, isChecked ->
            prefs.putBoolean(R.string.prefkey_content_album_art, isChecked)
        }

        // Insert button
        binding.insertButton.setOnClickListener {
            val dialogFragment = InsertPropertyDialogFragment.newInstance()
            dialogFragment.itemSelectListener = object:InsertPropertyDialogFragment.ItemSelectListener {
                override fun onItemSelect(insertString: String) {
                    var text = insertString
                    val index1 = binding.contentEditText.selectionStart
                    val index2 = binding.contentEditText.selectionEnd
                    val start = Math.min(index1, index2)
                    val end = Math.min(index1, index2)
                    val editable = binding.contentEditText.text
                    if (start > 0 && editable[start - 1] != ' ') {
                        text = " $text" // スペース挿入
                    }
                    editable.replace(start, end, text)
                }
            }
            dialogFragment.show(this@EditActivity)
        }

        // Priority
        binding.priorityButton.setOnClickListener {
            val dialogFragment = PropertyPriorityDialogFragment.newInstance()
            dialogFragment.clickListener = listener@{ _, _, _ -> }
            dialogFragment.show(this@EditActivity)
        }

        // Initialize
        binding.initializeButton.setOnClickListener {
            val dialogFragment = ConfirmDialogFragment.newInstance(getString(R.string.message_confirm_initialize_format), getString(R.string.label_confirm))
            dialogFragment.clickListener = listener@{ _, which, _ ->
                if (which == DialogInterface.BUTTON_POSITIVE) {
                    binding.contentEditText.setText(getString(R.string.format_content_default))
                    binding.insertAlbumArtCheckBox.isChecked = true
                }
            }
            dialogFragment.show(this@EditActivity)
        }

        binding.contentEditText.setText(prefs.getString(R.string.prefkey_content_format, defRes = R.string.format_content_default))
        binding.insertAlbumArtCheckBox.isChecked = prefs.getBoolean(R.string.prefkey_content_album_art, true)
    }

    /**
     * onStop
     */
    public override fun onStop() {
        super.onStop()

        prefs.putString(R.string.prefkey_content_format, binding.contentEditText.text)
    }

    /**
     * onOptionsItemSelected
     */
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }

        return super.onOptionsItemSelected(item)
    }

}
